File: src/main/java/ac/dnd/server/admission/domain/model/ApplicantData.java
package ac.dnd.server.admission.domain.model;

import java.time.LocalDateTime;

import ac.dnd.server.enums.ApplicantStatus;
import ac.dnd.server.enums.ApplicantType;

public record ApplicantData(
	Long id,
	String eventName,
	String name,
	String email,
	ApplicantType type,
	ApplicantStatus status,
	ViewablePeriod period
) {

	public boolean isViewable(
		final LocalDateTime now
	) {
		return period.contains(now);
	}
}

File: src/main/kotlin/ac/dnd/server/admission/domain/model/ApplicantData.kt
package ac.dnd.server.admission.domain.model

import ac.dnd.server.admission.domain.enums.ApplicantStatus
import ac.dnd.server.admission.domain.enums.ApplicantType
import java.time.LocalDateTime

data class ApplicantData(
    val id: Long,
    val eventName: String,
    val name: String,
    val email: String,
    val type: ApplicantType,
    val status: ApplicantStatus,
    val period: ViewablePeriod
) {
    fun isViewable(now: LocalDateTime): Boolean {
        return period.contains(now)
    }
}
